From b33a22b91748f14e728716b8297e26c4076faf23 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Fri, 6 Aug 2021 16:27:27 +0530
Subject: [PATCH 1895/1921] octeontx2-af: cn10K: modify default vWQE timer
 interval

Set the NIX AF vWQE timer interval to 0x3F which translates to 6400ns
This allows NIX LFs to configure vWQE timeout between 6.4us to 1.6ms.

Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Change-Id: I8fcd306e000bdf4109114b4c712d4b8fcb37bcfc
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/58087
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cn10k.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cn10k.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cn10k.c
index 7d29b7f73675..ac99a00a2c37 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cn10k.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cn10k.c
@@ -552,8 +552,10 @@ void rvu_nix_block_cn10k_init(struct rvu *rvu, struct nix_hw *nix_hw)
 {
 	int blkaddr = nix_hw->blkaddr;
 
-	/* Set vWQE timer interval to max possible value i.e. 102us. */
-	rvu_write64(rvu, blkaddr, NIX_AF_VWQE_TIMER, 0x3FFULL);
+	/* Set AF vWQE timer interval to a LF configurable range of
+	 * 6.4us to 1.632ms.
+	 */
+	rvu_write64(rvu, blkaddr, NIX_AF_VWQE_TIMER, 0x3FULL);
 }
 
 void rvu_apr_block_cn10k_init(struct rvu *rvu)
-- 
2.31.1

